src/main.cpp

```diff
@@ -1348,7 +1348,7 @@ void vcRenderSceneWindow(vcState *pProgramState)
     static bool wasContextMenuOpenLastFrame = false;
     bool selectItem = (io.MouseDragMaxDistanceSqr[0] < (io.MouseDragThreshold*io.MouseDragThreshold)) && ImGui::IsMouseReleased(0) && ImGui::IsItemHovered();
  
-    if (io.MouseDragMaxDistanceSqr[1] < (io.MouseDragThreshold*io.MouseDragThreshold) && ImGui::BeginPopupContextItem("SceneContext"))
+    if (pProgramState->settings.camera.cameraMouseBindings[1] != vcCPM_Forward && (io.MouseDragMaxDistanceSqr[1] < (io.MouseDragThreshold*io.MouseDragThreshold) && ImGui::BeginPopupContextItem("SceneContext")))
```

Comment:
I think swapping the drag threshold for a time threshold would be better. If the user right clicks for less than 100ms or something it opens the context menu, otherwise it does the currently mapped feature.